engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java

```diff
@@ -85,13 +90,34 @@ private Cache(
    *     compute the hash of the stored cache and compare it with the stored metadata entry.
    */
   static <T, M> Cache<T, M> create(
-      Cache.Spi<T, M> spi,
+      Spi<T, M> spi,
       Level logLevel,
       String logName,
       boolean needsSourceDigestVerification,
       boolean needsDataDigestVerification) {
     return new Cache<>(
-        spi, logLevel, logName, needsSourceDigestVerification, needsDataDigestVerification);
+        spi,
+        logLevel,
+        logName,
+        needsSourceDigestVerification,
+        needsDataDigestVerification,
+        Arena.ofShared());
+  }
+
+  static <T, M> Cache<T, M> create(
+      Spi<T, M> spi,
+      Level logLevel,
+      String logName,
+      boolean needsSourceDigestVerification,
+      boolean needsDataDigestVerification,
+      Arena memoryArena) {
```

Comment:
I don't think we want to provide an `Arena` from outside. It is an internal thing - a new arena should be allocated when mmapping a file and closed before writing.